start-build-env.sh

```diff
@@ -92,14 +92,26 @@ cd "${ROOTDIR}"
 # release managers can use the container to do releases
 
 dockerargs=(--rm=true)
-dockerargs+=(-v "${PWD}:/home/${USER_NAME}/yetus${V_OPTS:-}")
 dockerargs+=(-w "/home/${USER_NAME}/yetus")
+dockerargs+=(-v "${PWD}:/home/${USER_NAME}/yetus${V_OPTS:-}")
+
+# maven cache
+if [[ ! -d  ${HOME}/.m2 ]]; then
+  mkdir "${HOME}/.m2"
+fi
+dockerargs+=(-v "${HOME}/.m2:/home/${USER_NAME}/.m2${V_OPTS:-}")
+
+# GPG Signing for dist creation
+if [[ ! -d ${HOME}/.gnupg ]]; then
+  mkdir "${HOME}/.gnupg"
+fi
+dockerargs+=(-v "${HOME}/.gnupg:/home/${USER_NAME}/.gnupg${V_OPTS:-}")
 
-if [[ -z ${GITHUB_ACTIONS} ]]; then
-  dockerargs+=(-v "${HOME}/.m2:/home/${USER_NAME}/.m2${V_OPTS:-}")
-  dockerargs+=(-v "${HOME}/.gnupg:/home/${USER_NAME}/.gnupg")
-  dockerargs+=(-v "${HOME}/.ssh:/home/${USER_NAME}/.ssh")
+# git opertions
+if [[ ! -d ${HOME}/.ssh ]]; then
+  mkdir "${HOME}/.ssh"
 fi
+dockerargs+=(-v "${HOME}/.ssh:/home/${USER_NAME}/.ssh${V_OPTS:-}")
```

Comment:
There's not a better way to pass key materials into the docker process? I'd hope for some solution other than making the private keys visible to the process, something like agent forwarding.

Reply:
There are a lot of games one can play with the gpg-agent socket. We should probably look into them now that ssh login into github, etc, are more prevalent.  It should be noted that this code is in start-build-env.sh, which up until now was only used for local work.  In the GitHub actions case, there are no ssh creds to utilize.